Source/Public/Publish-Bicep.ps1

```diff
@@ -1,14 +1,30 @@
 function Publish-Bicep {
     [CmdletBinding()]
     param (
-        [Parameter(Mandatory, Position = 1)]
+        [Parameter(ParameterSetName = 'Default', Mandatory, Position = 1)]
+        [Parameter(ParameterSetName = 'Registry', Position = 1)]
         [ValidateNotNullOrEmpty()]
-        [ValidateScript({Test-Path $_})]
+        [ValidateScript( { Test-Path $_ })]
         [string]$Path,
 
-        [Parameter(Mandatory, Position = 2)]
+        [Parameter(ParameterSetName = 'Default', Position = 2)]
         [ValidateNotNullOrEmpty()]
-        [string]$Target          
+        [string]$Target,
+        
+        [Parameter(ParameterSetName = 'Registry', Mandatory, Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Registry,
+
+        [Parameter(ParameterSetName = 'Registry', Mandatory, Position = 3)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript( { $_ -notlike '/*' },
+        ErrorMessage = "Repository must be defined without a leading /")] 
+        [string]$Repository,
+
+        [Parameter(ParameterSetName = 'Registry', Mandatory, Position = 4)]
+        [ValidateNotNullOrEmpty()]
```

Comment:
This comment doesn't really belong here, but I dont know where to put it otherwise.. Just like with repository we can add a `$tag = $tag.TrimStart(':')` in the begin block to support input of  tag with or without leading ':'